expression/builtin_math.go

```diff
@@ -457,7 +457,6 @@ func (b *builtinCeilDecToIntSig) evalInt(row []types.Datum) (int64, bool, error)
 	// err here will only be ErrOverFlow(will never happen) or ErrTruncate(can be ignored).
 	res, err := val.ToInt()
 	if err == types.ErrTruncated && !val.IsNegative() {
-		err = nil
```

Comment:
seems we should not remove this,
but modify line 463 to `return res, false, errors.Trace(err)`